python/src/deltachat/chat.py

```diff
@@ -416,7 +417,7 @@ def get_messages(self):
         """
         dc_array = ffi.gc(
```

Comment:
Let's say `dc_get_chat_msgs` returns `NULL` here. It goes inside `ffi.gc` and then gets unreferenced with `safe_unref` later. You successfully avoided pasing `NULL` into `dc_array_unref`, but the problem still remains: `dc_array` is invalid and is used in other functions.

What is the meaning of creating such a `NULL` dc_array object and putting it inside of `iter_array`? It will call `dc_array_get_cnt` on `NULL` anyway, and print "careless call" warnings.

IMO if `dc_get_chat_msgs` returns `NULL`, an exception should be thrown as soon as possible to avoid running the rest of the code on the invalid values.

Reply:
yeah I tried just to avoid calling the unref on NULL but the real issue is that the bindings are not checking for NULL values, I don't even know if all that calls can return NULL, the one I was having issues with was `dc_context_unref` when passing a db path that core didn't have permission to write.

Reply:
so I need some core dev to tell if NULL was not handled there because that functions never return NULL, and then the only ones that return NULL are `dc_context_new` and `dc_context_new_closed` ???

Reply:
So, `dc_get_chat_msgs()` returns NULL only if the first parameter was NULL; `Box::into_raw()` never returns NULL, and the `block_on(async move {` doesn't change its result:

https://github.com/deltachat/deltachat-core-rust/blob/7598c50dbaa2abcbd417d96a02743269f666597b/deltachat-ffi/src/lib.rs#L1080-L1093

`dc_get_chat()` returns NULL also on database errors:

https://github.com/deltachat/deltachat-core-rust/blob/7598c50dbaa2abcbd417d96a02743269f666597b/deltachat-ffi/src/lib.rs#L1448

Reply:
Does this help?